php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCompletionItem.java

```diff
@@ -134,10 +134,11 @@ public abstract class PHPCompletionItem implements CompletionProposal {
     private QualifiedNameKind generateAs;
     private static ScheduledExecutorService service = Executors.newSingleThreadScheduledExecutor();
     private static final Cache<FileObject, PhpLanguageProperties> PROPERTIES_CACHE
-            = new Cache<>(new WeakHashMap<FileObject, PhpLanguageProperties>());
+            = new Cache<>(new WeakHashMap<>());
     private final boolean isPlatform;
     private final boolean isDeprecated;
     private PhpVersion phpVersion;
+    private static Boolean ADD_FIRST_CLASS_CALLABLE = null; // for unit tests
```

Comment:
I think this field should be `volatile`, no? Or is always read/written in just one thread? If so, please document it. Thank you.

Reply:
Will fix it. Thank you!

Reply:
Fixed https://github.com/apache/netbeans/compare/7069e4d2f2b90e385bc61681e1b8e4b37770bb24..34af956e74b9b92809cefdfd1126005d8ba35697